src/e2e/java/uk/gov/hmcts/reform/sscs/functional/ccd/PipDecisionNoticeFunctionalTest.java

```diff
@@ -0,0 +1,1183 @@
+            new Object[] {"telephone", false, false},
+            new Object[] {"telephone", false, true},
+            new Object[] {"telephone", true, false},
+            new Object[] {"telephone", true, true},
+            new Object[] {"triage", false, false},
+            new Object[] {"triage", false, true},
+            new Object[] {"triage", true, false},
+            new Object[] {"triage", true, true},
+            new Object[] {"faceToFace", false, false},
+            new Object[] {"faceToFace", false, true},
+            new Object[] {"faceToFace", true, false},
+            new Object[] {"faceToFace", true, true},
+        };
+    }
+
+    @Test
+    @Parameters(named = "allowed")
+    public void nonDescriptorFlow_shouldGeneratePdfWithExpectedText(boolean allowed) throws IOException {
+
+        String json = getJsonCallbackForTestAndReplace("handlers/writefinaldecision/pipScenarioCallbackNonDescriptorFlow.json", Arrays.asList("ALLOWED_OR_REFUSED"),
+            Arrays.asList(allowed ? "allowed" : "refused"));
+
+        byte[] bytes = callPreviewFinalDecision(json);
+        try (PDDocument document = PDDocument.load(bytes)) {
+            String pdfText = new PDFTextStripper().getText(document);
+            String pdfTextWithoutNewLines = replaceNewLines(pdfText);
+            if (allowed) {
+                assertThat(pdfTextWithoutNewLines, containsString("1. The appeal is allowed."));
+                assertThat(pdfTextWithoutNewLines, containsString("2. The decision made by the Secretary of State on 17/11/2020 in respect of Personal Independence Payment is set aside."));
+            } else {
+                assertThat(pdfTextWithoutNewLines, containsString("1. The appeal is refused."));
+                assertThat(pdfTextWithoutNewLines, containsString("2. The decision made by the Secretary of State on 17/11/2020 in respect of Personal Independence Payment is confirmed."));
+            }
+            assertThat(pdfTextWithoutNewLines, containsString("3. My summary."));
+            assertThat(pdfTextWithoutNewLines, containsString("4. Reasons for decision 1"));
+            assertThat(pdfTextWithoutNewLines, containsString("5. Reasons for decision 2"));
+            assertThat(pdfTextWithoutNewLines, containsString("6. Anything else."));
+            assertThat(pdfTextWithoutNewLines, containsString(
+                "7. This has been a remote hearing in the form of a video hearing. Joe Bloggs attended the hearing today and the tribunal considered the appeal bundle to page B7. A Presenting Officer attended on behalf of the Respondent."));
+            assertThat(pdfTextWithoutNewLines, not(containsString("8.")));
+        }
+    }
+
+    @Test
+    @Parameters(named = "hearingTypeCombinations")
+    public void notConsideredNoAward_shouldGeneratePdfWithExpectedTextForHearingType(String hearingType, boolean appellantAttended, boolean presentingOfficerAttended) throws IOException {
```

Comment:
This test has very complex logic. I guess it needs to be so since the functionality is very complex.